### PR TITLE
#34 admin crud operations & assembled admin panel

### PR DESCRIPTION
**Pull request by artempenteskul**

### CODE REVIEW

File: admin/app.py [hunk 1]
```diff
@@ -152,6 +160,125 @@ def logout():
     logout_user()
     return redirect(url_for('start_page'))
 
+@login.user_loader
```
Comment (Palusha) [inline admin/app.py:160]: @tommyfromche A duplicate of the method already at the bottom

File: admin/app.py [hunk 2]
```diff
@@ -152,6 +160,125 @@ def logout():
...
+        book.language = form.language.data
+        book.description = form.description.data
+        book.pages = form.pages.data
+        book.genres = form.genre.data
+        book.status = 'active'
+        try:
+            book.save()
+            return redirect('/book-storage')
+        except Exception as e:
+            traceback.print_exc(e)
```
Comment (ivan-dubei) [inline admin/app.py]: Add message to the UI for the admin with additional info about failure.

File: admin/app.py [hunk 3]
```diff
@@ -11,10 +12,18 @@
...
 
+from forms import LoginForm, AddUserForm, UpdateUserForm, AddBookForm, UpdateBookForm
+from models import User, Book
+
+
 load_dotenv()
 
 app = Flask(__name__)
 app.config['SECRET_KEY'] = os.getenv('SECRET_KEY')
+UPLOAD_FOLDER = 'static/files'
```
Comment (ivan-dubei) [inline admin/app.py]: I guess that we need to put this into config file

File: admin/app.py [hunk 4]
```diff
@@ -152,6 +160,125 @@ def logout():
...
+            publisher = form.publisher.data
+            language = form.language.data
+            description = form.description.data
+            pages = form.pages.data
+            genres = form.genre.data
+            status = form.status.data
+            book.update(title=title, author=author, year=year, publisher=publisher, language=language,
+                        description=description, pages=pages, genres=genres, status=status)
+            return redirect('/book-storage')
+    except Exception as e:
```
Comment (ivan-dubei) [inline admin/app.py:261]: we need some useful message on the UI with the reason of failure, I assume that printing traceback in the server log is insufficient

File: admin/app.py [hunk 5]
```diff
@@ -152,6 +160,125 @@ def logout():
...
+
+
+@app.route('/book-delete/<string:_id>')
+def book_delete(_id):
+    try:
+        book = Book.objects.get(id=ObjectId(_id), status='active')
+        book.update(status='inactive')
+        return redirect('/book-active')
+    except Exception as e:
+        traceback.print_exc(e)
```
Comment (ivan-dubei) [inline admin/app.py]: the same case here, add some message on the UI

File: admin/app.py [hunk 6]
```diff
@@ -152,6 +160,125 @@ def logout():
...
+        return redirect('/home')
+
+
+@app.route('/book-restore/<string:_id>')
+def book_restore(_id):
+    try:
+        book = Book.objects.get(id=ObjectId(_id), status='inactive')
+        book.update(status='active')
+        return redirect('/book-inactive')
+    except Exception as e:
```
Comment (ivan-dubei) [inline admin/app.py:288]: add some message on the UI

File: admin/forms.py [hunk 7]
```diff
@@ -1,5 +1,5 @@
 from flask_wtf import FlaskForm
-from wtforms import StringField, SelectField, PasswordField, SubmitField
+from wtforms import StringField, SelectField, PasswordField, SubmitField, IntegerField
```
Comment (ivan-dubei) [inline admin/forms.py]: we need alphabetical order for imports

**Review by Palusha — APPROVED** [review]

**Review by ivan-dubei — APPROVED** [review]

**Pull request merged**